File: NativeObjc/NativeObjc/ViewController.m
//
//  ViewController.m
//  NativeObjc
//
//  Created by WEN WEI LIN on 2020/8/13.
//  Copyright © 2020 WEN WEI LIN. All rights reserved.
//

#import "ViewController.h"
#import "MakarViewerManager.h"
#import "AppDelegate.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[MakarViewerManager shared] setDelegate:self];
}

- (IBAction)btnEnterAR:(id)sender
{
    NSString *projectId = @"2e011aec1f0be6547f9069bfe05816ee";
    [[MakarViewerManager shared] showProjectWithProjectId:projectId type: AR];
}

- (IBAction)btnEnterARSlam:(id)sender
{
    NSString *projectId = @"ca6e231b9bde8183dc8f208ab619055a";
    [[MakarViewerManager shared] showProjectWithProjectId:projectId type: AR_SLAM];
}

- (IBAction)btnEnterVr:(id)sender
{
    NSString *projectId = @"a1d05d718e2ba6991806e0698ceaab34";
    [[MakarViewerManager shared] showProjectWithProjectId:projectId type: VR];
}

//MARK: - MakarViewerDelegate
- (void)makarDidLoad {
    NSLog(@"Receive Makar did load");
}

- (void)makarDidUnload {
    NSLog(@"Receive Makar did unload");
}

- (void)makarDidQuit {
    NSLog(@"Receive Makar did quit");
}

-(void)makarRequestOrientation:(NSString *)orientation{
    AppDelegate *appDelegate = (AppDelegate *) [[UIApplication sharedApplication] delegate];
    if([orientation isEqualToString:@"portrait"]){
        [appDelegate setOrientationLock:UIInterfaceOrientationMaskPortrait];
    }else{
        [appDelegate setOrientationLock:UIInterfaceOrientationMaskLandscape];
    }
    
}
@end

File: NativeSwift/NativeSwift/ViewController.swift
//
//  ViewController.swift
//  NativeSwift
//
//  Created by WEN WEI LIN on 2021/2/2.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        MakarViewerManager.shared()?.delegate = self
    }
    
    @IBAction
    func btnEnterAR(){
        let projectId = "c2ca3fda-83fb-4a74-83ea-41bbe084e0a9"
        MakarViewerManager.shared()?.showProject(withProjectId: projectId, type: .AR)
    }

    @IBAction
    func btnEnterARSlam(){
        let projectId = "830f8ae9117aa2bc30097ba16936bbe9"
        MakarViewerManager.shared()?.showProject(withProjectId: projectId, type: .AR_SLAM)
    }
    
    @IBAction
    func btnEnterVR(){
        let projectId = "f6574a6c-4595-4124-8c9f-a9605a8b5d8c"
        MakarViewerManager.shared()?.showProject(withProjectId: projectId, type: .VR)
    }
    
    /*
    @IBAction
    func btnEnterHand(){
        MakarViewerManager.shared()?.showProject(withProjectId: PROJECT_ID, type: .HAND)
    }

    @IBAction
    func btnEnterMix(){
        MakarViewerManager.shared()?.showProject(withProjectId: PROJECT_ID, type: .MIX)
    }
     */

    func setOrientation(_ orientation: UIInterfaceOrientationMask){
        AppDelegate.AppUtility.lockOrientation(orientation, andRotateTo: UIInterfaceOrientation.portrait)
    }

}

extension ViewController: MakarViewerDelegate{
    
    func makarDidLoad() {
        //TODO
    }
    
    func makarDidUnload() {
        //TODO
    }
    
    func makarDidQuit() {
        //TODO
    }
        
    func makarRequestOrientation(_ orientation: String!) {
        //Makar project require device orientation
        let targetOrientation = orientation == "portrait" ? UIInterfaceOrientationMask.portrait: UIInterfaceOrientationMask.landscape
        setOrientation(targetOrientation)
    }


}

